modules/pxp-module-puppet.md

```diff
@@ -0,0 +1,212 @@
+
+### Error Types
+
+If a run fails the `"error_type"` field will be set to one of the following values:
+
+- `"invalid_json"` pxp-module-puppet was called with invalid json on stdin
+- `"no_puppet_bin"` The executable specified by `puppet_bin` doesn't exist
+- `"no_last_run_report"` last_run_report.yaml doesn't exist
+- `"invalid_last_run_report"` last_run_report.yaml could not be parsed
+- `"agent_already_running"` Puppet agent is already performing a run
+- `"agent_disabled"` Puppet agent is disabled
+- `"agent_failed_to_start"` Puppet agent failed to start
+- `"agent_exit_non_zero"` Puppet agent exited with a non-zero exitcode
+
+### Example error responses
+
+
+### The `puppet_bin` file does not exist
+
+```
+{
+    "kind" : "unknown",
+    "time" : "unknown",
+    "transaction_uuid" : "unknown",
+    "environment" : "unknown",
+    "status" : "unknown",
+    "error_type" : "no_puppet_bin",
+    "error" : "Puppet executable '$puppet_bin' does not exist",
+    "exitcode" : -1
+}
+```
+
+### The Puppet agent is busy performing a run when the run action is called
+
+```
+{
+    "kind" : "unknown",
+    "time" : "unknown",
+    "transaction_uuid" : "unknown",
+    "environment" : "unknown",
+    "status" : "unknown",
+    "error_type" : "agent_already_running",
+    "error" : "Puppet is already running",
+    "exitcode" : -1
+}
+```
+
+### The Puppet agent has been disabled on the host the run action was called on
+
+```
+{
+    "kind" : "unknown",
+    "time" : "unknown",
+    "transaction_uuid" : "unknown",
+    "environment" : "unknown",
+    "status" : "unknown",
+    "error_type" : "agent_disabled",
+    "error" : "Puppet is disabled",
+    "exitcode" : -1
+}
+```
+
+### The run action was called with invalid JSON on stdin
+
+```
+{
+    "kind" : "unknown",
+    "time" : "unknown",
+    "transaction_uuid" : "unknown",
+    "environment" : "unknown",
+    "status" : "unknown",
+    "error_type" : "invalid_json",
+    "error" : "Invalid json parsed on STDIN",
+    "exitcode" : -1
+}
+```
+
+### The Puppet run failed to start
+
+```
+{
+    "kind" : "unknown",
+    "time" : "unknown",
+    "transaction_uuid" : "unknown",
+    "environment" : "unknown",
+    "status" : "unknown",
+    "error_type" : "agent_failed_to_start",
+    "error" : "Failed to start Puppet",
+    "exitcode" : -1
+}
+```
+
+### Puppet exited with a non-zero exitcode
+
+```
+{
+    "kind" : "unknown",
+    "time" : "unknown",
+    "transaction_uuid" : "unknown",
+    "environment" : "unknown",
+    "status" : "unknown",
+    "error_type" : "agent_exit_non_zero",
+    "error" : "Puppet exited with a non 0 exitcode",
+    "exitcode" : $exitcode
+}
+```
+
+### last_run_report.yaml doesn't exist after a successful run
+
+```
+{
+    "kind" : "unknown",
+    "time" : "unknown",
+    "transaction_uuid" : "unknown",
+    "environment" : "unknown",
+    "status" : "unknown",
+    "error_type" : "no_last_run_report",
+    "error" : "$last_run_report.yaml doesn't exist",
```

Comment:
I suggest to revisit this documentation once pxp-module-puppet is updated and not touch it now.